ansible/roles/test/files/ptftests/advanced-reboot.py

```diff
@@ -787,6 +786,56 @@ def pre_reboot_test_setup(self):
         watcher = self.pool.apply_async(self.reachability_watcher)
         time.sleep(5)
 
+    def get_warmboot_finalizer_state(self):
+        stdout, stderr, _ = self.dut_connection.execCommand('sudo systemctl is-active warmboot-finalizer.service')
+
+        if stdout == []:
+            self.log('Finalizer state not returned from DUT')
+            #self.fails['dut'].add('Error collecting Finalizer state from DUT')
+            return ''
+        if stderr != []:
+            self.fails['dut'].add("stderr from DUT while collecting Finalizer state: %s" % (str(stderr)))
+
+        finalizer_state = stdout[0].strip()
+        return finalizer_state
+
+    def get_now_time(self):
+        stdout, stderr, _ = self.dut_connection.execCommand('date +"%Y-%m-%d %H:%M:%S"')
+        if stdout == []:
+            self.fails['dut'].add('Error collecting current date from DUT: empty value returned')
+        if stderr != []:
+            self.fails['dut'].add("Error collecting current date from DUT: %s" % (str(stderr)))
+        return datetime.datetime.strptime(stdout[0].strip(), "%Y-%m-%d %H:%M:%S")
+
+    def check_warmboot_finalizer(self):
+        dut_datetime = self.get_now_time()
+        self.log('waiting for warmboot-finalizer service to become activating')
+        finalizer_state = self.get_warmboot_finalizer_state()
```

Comment:
if this method returns '', do you want to handle it and skip the rest of the checks?

Reply:
Actually in very early stages, if this is empty I still want to keep checking for the `finalizer_timeout` period. If it continues to be empty, an exception will eventually be raised after `finalizer_timeout` seconds.